LanguageExt.Core/DataTypes/UnitsOfMeasure/ScaleTemp.cs

```diff
@@ -0,0 +1,170 @@
+    /// type.  So keep all ScaleTemps wrapped until you need the
+    /// value, then extract using various unit-of-measure
+    /// accessors (Kelvin, etc.) or divide by 1.Celsius()
+    /// </summary>
+    public struct ScaleTemp :
+        IComparable<ScaleTemp>,
+        IEquatable<ScaleTemp>
+    {
+        readonly double Value;
+
+        internal ScaleTemp(double scaleTemp)
+        {
+            if (scaleTemp < UnitsAbsTempExtensions.AbsZeroInDegC)
+                throw new ArgumentOutOfRangeException(nameof(scaleTemp));
+
+            Value = scaleTemp;
+        }
+
+        public override string ToString() =>
+            Value + " °C";
+
+        public bool Equals(ScaleTemp other) =>
+            Value.Equals(other.Value);
+
+        public bool Equals(ScaleTemp other, double epsilon) =>
+            Math.Abs(other.Value - Value) < epsilon;
+
+        public bool Equals(AbsTemp other) =>
+            Value.Equals(other.Celsius);
+
+        public override bool Equals(object obj) =>
+            obj is ScaleTemp st
+            ? Equals(st)
+            : false;
+
+        public override int GetHashCode() =>
+            Value.GetHashCode();
+
+        public int CompareTo(ScaleTemp other) =>
+            Value.CompareTo(other.Value);
+
+        public ScaleTemp Append(ScaleTemp rhs) =>
+            new ScaleTemp(Value + rhs.Value);
+
+        public ScaleTemp Subtract(ScaleTemp rhs) =>
+            new ScaleTemp(Value - rhs.Value);
+
+        public ScaleTemp Multiply(double rhs) =>
+            new ScaleTemp(Value * rhs);
+
+        public ScaleTemp Divide(double rhs) =>
+            new ScaleTemp(Value / rhs);
+
+        public static ScaleTemp operator *(ScaleTemp lhs, double rhs) =>
+            lhs.Multiply(rhs);
+
+        public static ScaleTemp operator *(double lhs, ScaleTemp rhs) =>
+            rhs.Multiply(lhs);
+
+        public static ScaleTemp operator +(ScaleTemp lhs, ScaleTemp rhs) =>
+            lhs.Append(rhs);
+
+        public static ScaleTemp operator -(ScaleTemp lhs, ScaleTemp rhs) =>
+            lhs.Subtract(rhs);
+
+        public static ScaleTemp operator /(ScaleTemp lhs, double rhs) =>
+            lhs.Divide(rhs);
+
+        public static double operator /(ScaleTemp lhs, ScaleTemp rhs) =>
+            lhs.Value / rhs.Value;
+
+        public static bool operator ==(ScaleTemp lhs, ScaleTemp rhs) =>
+            lhs.Equals(rhs);
+
+        public static bool operator ==(ScaleTemp lhs, AbsTemp rhs) =>
+            lhs.Equals(rhs);
+
+        public static bool operator !=(ScaleTemp lhs, ScaleTemp rhs) =>
+            !lhs.Equals(rhs);
+
+        public static bool operator !=(ScaleTemp lhs, AbsTemp rhs) =>
+            !lhs.Equals(rhs);
+
+        public static bool operator >(ScaleTemp lhs, ScaleTemp rhs) =>
+            lhs.Value > rhs.Value;
+
+        public static bool operator >(ScaleTemp lhs, AbsTemp rhs) =>
+            lhs.Celsius > rhs.Celsius;
```

Comment:
This is inconsistent with the AbsTemp implementation, we should be doing `lhs.Kelvin > rhs.Kelvin` to make sure any precision loss is identical. Same with the other `operator` methods mixing `ScaleTemp` and `AbsTemp`